test/functional/wallet_decommission_request.py

```diff
@@ -0,0 +1,334 @@
+                                },
+                                "index": 1,
+                            },
+                        },
+                ],
+                 "kernel_input_utxo": [
+                    {
+                        "CreateStakePool": [
+                            self.genesis_pool_id(),
+                            {
+                                "value": 40_000*ATOMS_PER_COIN,
+                                "staker": {
+                                    "PublicKey": self.public_key(GENESIS_STAKE_PUBLIC_KEY),
+                                },
+                                "vrf_public_key": self.vrf_public_key(GENESIS_VRF_PUBLIC_KEY),
+                                "decommission_key": {
+                                    "PublicKey": self.public_key(GENESIS_STAKE_PUBLIC_KEY),
+                                },
+                                "margin_ratio_per_thousand": 1000,
+                                "cost_per_block" : "0"
+                            },
+                        ],
+                    }
+                ],
+            }
+        }).to_hex()[2:]
+
+        self.generate_block(1, block_input_data, transactions)
+
+    def gen_pos_block(self, transactions, block_height, block_id=None):
+        block_id = self.previous_block_id() if block_id is None else block_id
+        block_input_data = block_input_data_obj.encode({
+            "PoS": {
+                "stake_private_key": self.private_key(GENESIS_STAKE_PRIVATE_KEY),
+                "vrf_private_key": self.vrf_private_key(GENESIS_VRF_PRIVATE_KEY),
+                "pool_id": self.genesis_pool_id(),
+                "kernel_inputs": [
+                        {
+                            "Utxo": {
+                                "id": {
+                                    "BlockReward": block_id
+                                },
+                                "index": 0,
+                            },
+                        },
+                ],
+                 "kernel_input_utxo": [
+                    {
+                        "ProduceBlockFromStake": [
+                            {
+                                "PublicKey": self.public_key(GENESIS_STAKE_PUBLIC_KEY),
+                            },
+                            self.genesis_pool_id(),
+                        ],
+                    }
+                ],
+            }
+        }).to_hex()[2:]
+
+        self.generate_block(block_height, block_input_data, transactions)
+
+    async def async_test(self):
+        node = self.nodes[0]
+        decommission_pub_key_encoded = ""
+
+        async with WalletCliController(node, self.config, self.log, chain_config_args=["--chain-pos-netupgrades", "true"]) as wallet:
+            # new cold wallet
+            await wallet.create_wallet("cold_wallet")
+
+            decommission_pub_key_bytes = await wallet.new_public_key()
```

Comment:
Please use addresses. No need to use public keys anymore. This function will be soon removed, btw.